src/widgets/configure/configure.ts

```diff
@@ -1,15 +1,25 @@
 import { PlainSearchParameters } from 'algoliasearch-helper';
-import connectConfigure from '../../connectors/configure/connectConfigure';
-import { WidgetFactory } from '../../types';
+import connectConfigure, {
+  ConfigureRendererOptions,
+  ConfigureConnectorParams,
+} from '../../connectors/configure/connectConfigure';
+import { Widget, WidgetRenderState } from '../../types';
 import { noop } from '../../lib/utils';
 
 /**
  * A list of [search parameters](https://www.algolia.com/doc/api-reference/search-api-parameters/)
  * to enable when the widget mounts.
  */
-export type ConfigureWidgetOptions = PlainSearchParameters;
+export type ConfigureWidgetParams = PlainSearchParameters;
 
-export type ConfigureWidget = WidgetFactory<{}, ConfigureWidgetOptions>;
+export type ConfigureWidget = (
+  widgetParams: ConfigureConnectorParams['searchParameters']
+) => Widget<{
+  renderState: WidgetRenderState<
```

Comment:
why is this different to the others?

Reply:
I forgot to mention it. If it was

```js
export type ConfigureWidget = WidgetFactory<
  ConfigureRendererOptions,
  ConfigureConnectorParams,
  ConfigureWidgetParams
>;
```

(just like others)

then users would've done something like

```js
configure({ searchParameters: { ... });
```

but inside the widget, we are actually passing `widgetParams` to `searchParameters`

```js
const configure: ConfigureWidget = function configure(widgetParams) {
  // This is a renderless widget that falls back to the connector's
  // noop render and unmount functions.
  const makeWidget = connectConfigure(noop);

  return makeWidget({ searchParameters: widgetParams });
};
```

So `widgetParams` in configure widget is a kind of irregular.